tinygrad/runtime/ops_torch.py

```diff
@@ -27,4 +27,4 @@ def fromCPU(cls, x):
     buf = torch.from_numpy(x).requires_grad_(False).to(device)
     return cls(prod(x.shape), type_map[buf.dtype], buf)
   def toCPU(self): return self._buf.cpu().numpy()
-TorchBuffer = Interpreted(RawTorchBuffer, torch_fxn_for_op, from_underlying=lambda x: RawTorchBuffer(prod(x.shape), type_map[x.dtype], x))
+TorchBuffer = Interpreted(RawTorchBuffer, torch_fxn_for_op, from_underlying=lambda x: RawTorchBuffer(prod(x.shape), type_map[x.dtype], x), to_underlying=lambda x: x._buf if x._buf.dtype!=torch.bool else x._buf.long())
```

Comment:
I was happy with everything until this line. I'm not sure this is the right fix. Is this a separate issue from the above one? Maybe we just merge the above.

Reply:
Yeah, there were 2 issues. One is the casting behaviour (https://github.com/tinygrad/tinygrad/issues/1495), second is that torch can't do `-` on bools (https://github.com/tinygrad/tinygrad/issues/1524). Now this PR only fixes the casting behaviour issue. 

Don't you mind adding float64/double types for fixing np crash from https://github.com/tinygrad/tinygrad/issues/1495 to close the issue completely?

Reply:
Fine with that.